pkg/image/api/v1beta2/types.go

```diff
@@ -0,0 +1,144 @@
+package v1beta2
+
+import (
+	kapi "github.com/GoogleCloudPlatform/kubernetes/pkg/api/v1beta3"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/runtime"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/util"
+)
+
+// ImageList is a list of Image objects.
+type ImageList struct {
+	kapi.TypeMeta `json:",inline"`
+	kapi.ListMeta `json:"metadata,omitempty"`
+
+	Items []Image `json:"items"`
+}
+
+// Image is an immutable representation of a Docker image and metadata at a point in time.
+type Image struct {
+	kapi.TypeMeta   `json:",inline"`
+	kapi.ObjectMeta `json:"metadata,omitempty"`
+
+	// The string that can be used to pull this image.
+	DockerImageReference string `json:"dockerImageReference,omitempty"`
+	// Metadata about this image
+	DockerImageMetadata runtime.RawExtension `json:"dockerImageMetadata,omitempty"`
+	// This attribute conveys the version of the object, which if empty defaults to "1.0"
+	DockerImageMetadataVersion string `json:"dockerImageMetadataVersion,omitempty"`
+	// The raw JSON of the manifest
+	DockerImageManifest string `json:"dockerImageManifest,omitempty"`
+}
+
+// ImageStreamList is a list of ImageStream objects.
+type ImageStreamList struct {
+	kapi.TypeMeta `json:",inline"`
+	kapi.ListMeta `json:"metadata,omitempty"`
+
+	Items []ImageStream `json:"items"`
+}
+
+// ImageStream stores a mapping of tags to images, metadata overrides that are applied
+// when images are tagged in a stream, and an optional reference to a Docker image
+// repository on a registry.
+type ImageStream struct {
+	kapi.TypeMeta   `json:",inline"`
+	kapi.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec describes the desired state of this stream
+	Spec ImageStreamSpec `json:"spec"`
+	// Status describes the current state of this stream
+	Status ImageStreamStatus `json:"status,omitempty"`
+}
+
+// ImageStreamSpec represents options for ImageStreams.
+type ImageStreamSpec struct {
+	// Optional, if specified this stream is backed by a Docker repository on this server
+	DockerImageRepository string `json:"dockerImageRepository,omitempty"`
+	// Tags map arbitrary string values to specific image locators
+	Tags []NamedTagReference `json:"tags,omitempty"`
```

Comment:
I know we just created this as part of the ImageStream work, but I'm questioning if TagReference is the appropriate name for this type. It's entirely valid to have annotations for a tag without there being any "reference" part. Would TagSpec or something along those lines be more appropriate?